auth/src/main/java/org/aerogear/auth/AbstractRole.java

```diff
@@ -0,0 +1,37 @@
+package org.aerogear.auth;
+
+/**
+ * Base class for roles pojos.
+ */
+abstract class AbstractRole implements IRole {
+
+    /**
+     * Unique identifier for this role.
+     */
+    private final String roleID;
+
+    /**
+     * Role friendly name
+     */
+    private final String roleName;
+
+    /**
+     * Builds a role object
+     * @param roleID Role unique id
+     * @param roleName Role friendly name
+     */
+    AbstractRole(final String roleID, final String roleName) {
+        this.roleID = roleID;
```

Comment:
I don't think we will need a role ID, as Keycloak/OpenID doesn't have that concept. Role names on the IDM server will be unique.